airflow/models/taskinstance.py

```diff
@@ -163,12 +165,12 @@ def set_current_context(context: Context) -> Generator[Context, None, None]:
 
 
 def clear_task_instances(
-    tis,
-    session,
+    tis: List["TaskInstance"],
+    session: Session,
     activate_dag_runs=None,
     dag=None,
```

Comment:
```suggestion
    activate_dag_runs: None = None,
    dag: Optional[DAG] = None,
```

Reply:
Interesting, never considered using typing to hint at things being deprecated.